doc/source/serve/configure-serve-deployment.md

```diff
@@ -0,0 +1,139 @@
+- `health_check_timeout_s` - Duration in seconds, that replicas wait for a health check method to return before considering it as failed. Defaults to 30s.
+- `graceful_shutdown_wait_loop_s` - Duration that replicas wait until there is no more work to be done before shutting down. Defaults to 2s.
+- `graceful_shutdown_timeout_s` - Duration to wait for a replica to gracefully shut down before being forcefully killed. Defaults to 20s.
+- `is_driver_deployment` - [EXPERIMENTAL] when set, exactly one replica of this deployment runs on every node (like a daemon set).
+
+There are 3 ways of specifying parameters:
+
+  - In the `@serve.deployment` decorator -
+
+```{literalinclude} ../serve/doc_code/configure_serve_deployment/model_deployment.py
+:start-after: __deployment_start__
+:end-before: __deployment_end__
+:language: python
+```
+
+  - Through `options()` -
+
+```{literalinclude} ../serve/doc_code/configure_serve_deployment/model_deployment.py
+:start-after: __deployment_end__
+:end-before: __options_end__
+:language: python
+```
+
+  - Using the YAML [Serve Config file](serve-in-production-config-file) -
+
+```yaml
+applications:
+
+- name: app1
+
+  route_prefix: /
+
+  import_path: configure_serve:translator_app
+
+  runtime_env: {}
+
+  deployments:
+
+  - name: Translator
+    num_replicas: 2
+    max_concurrent_queries: 100
+    graceful_shutdown_wait_loop_s: 2.0
+    graceful_shutdown_timeout_s: 20.0
+    health_check_period_s: 10.0
+    health_check_timeout_s: 30.0
+    ray_actor_options:
+      num_cpus: 0.2
+      num_gpus: 0.0
+```
+
+## Overriding deployment settings
+
+The order of priority is (from highest to lowest):
+
+1. Serve Config file
+2. `.options()` call in python code referenced above
+3. `@serve.deployment` decorator in python code
+4. Serve defaults
+
+For example, if a deployment's `num_replicas` is specified in the config file and their graph code, Serve will use the config file's value. If it's only specified in the code, Serve will use the code value. If the user doesn't specify it anywhere, Serve will use a default (which is `num_replicas=1`).
+
+Keep in mind that this override order is applied separately to each individual parameter.
+For example, if a user has a deployment `ExampleDeployment` with the following decorator:
+
+```python
+@serve.deployment(
+    num_replicas=2,
+    max_concurrent_queries=15,
+)
+class ExampleDeployment:
+    ...
+```
+
+and the following config file:
+
+```yaml
+...
+
+deployments:
+
+    - name: ExampleDeployment
+      num_replicas: 5
+
+...
+```
+
+Serve will set `num_replicas=5`, using the config file value, and `max_concurrent_queries=15`, using the code value (since `max_concurrent_queries` wasn't specified in the config file). All other deployment settings use Serve defaults since the user didn't specify them in the code or the config.
```

Comment:
```suggestion
Serve sets `num_replicas=5`, using the config file value, and `max_concurrent_queries=15`, using the code value (because `max_concurrent_queries` wasn't specified in the config file). All other deployment settings use Serve defaults because the user didn't specify them in the code or the config.
```